tests/test_package.py

```diff
@@ -0,0 +1,219 @@
+import pytest
+import requests
+
+from dcos_test_utils.helpers import Url
+from dcos_test_utils.package import Package
+from helpers import MockEmitter, MockResponse
+
+
+@pytest.fixture
+def replay_session(monkeypatch):
+    """A mocked session that sends back the configured responses in
+    order.
+    """
+    mock_session = MockEmitter(list())
+
+    monkeypatch.setattr(requests, 'Session',
+                        lambda *args, **kwargs: mock_session)
+
+    yield mock_session
```

Comment:
This should probably be wrapped in a `try: finally` block so that we always get the debug cache after an exception is raised